Mathlib/Geometry/Manifold/PrincipalFiberBundle/PrincipalGBundle.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2025 Dominic Steinitz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dominic Steinitz
+-/
+
+import Mathlib.Geometry.Manifold.Algebra.LieGroup
+import Mathlib.LinearAlgebra.UnitaryGroup
+import Mathlib.Topology.FiberBundle.Basic
+
+/-!
+# Principal G-Bundles
+
+This file defines smooth principal bundles with Lie group actions on manifolds.
+
+## Main definitions
+
+* `SmoothLeftGAction`: An `n`-times continuously differentiable left action of a Lie group
+  on a manifold
+* `SmoothRightGAction`: An `n`-times continuously differentiable right action of a Lie group
+  on a manifold
+* `PrincipalBundleCore`: The core structure of a principal bundle, consisting of a fiber bundle
+  with a free and transitive group action on the fibers
+
+## Implementation notes
+
+Right actions are encoded using `MulAction (MulOpposite G)` to maintain compatibility with
+Mathlib's action conventions while preserving the correct associativity for principal bundles.
+
+## References
+
+* Standard reference on principal bundles and connections
+
+-/
+
+open Matrix Bundle Manifold MulAction
+
+open RightActions
+
+/-- An `n`-times continuously differentiable left action of a Lie group `G` on a manifold `M`.
+
+This class asserts that the left action map `(g, m) ↦ g • m` is `n`-times continuously
+differentiable as a map `G × M → M` between manifolds. This is the standard notion of
+a Lie group acting differentiably on a manifold.
+
+## Smoothness parameter
+- `n`: The differentiability degree used uniformly for the Lie group `G`, the manifold `M`,
+  and the action map itself (Cⁿ structures throughout)
+- Common choices: `n = ⊤` (infinitely differentiable/smooth), or `n = ∞` (real analytic)
+
+The dimensions of `G` and `M` are independent and are determined by the model spaces
+`E_G` and `E_M` respectively.
+
+## Main condition
+- `smooth_smul`: The action map `(g, m) ↦ g • m` is `n`-times continuously differentiable
+  (ContMDiff with degree `n`)
+
+## Terminology note
+The field is named `smooth_smul` following Mathlib conventions, even though it asserts
+`n`-times differentiability rather than infinite differentiability when `n ≠ ⊤`.
+
+## Related definitions
+- For right actions, see `SmoothRightGAction`
+- For actions on principal bundles, the differentiability is typically automatic from the
+  fiber bundle structure
+-/
+class SmoothLeftGAction
+{𝕜 : Type*} [NontriviallyNormedField 𝕜] (n : WithTop ℕ∞)
+{E_G : Type*} [NormedAddCommGroup E_G] [NormedSpace 𝕜 E_G]
```

Comment:
Nit: mathlib convention is to use `E, H, I` for a manifold M, and perhaps `F, G, J` for a manifold `N`.
Can you check other files as needed, and adjust your names to match?